android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingViewModel.kt

```diff
@@ -179,4 +191,63 @@ constructor(
   fun dismissInsightsView() {
     viewModelScope.launch { unsyncedResourcesMutableSharedFlow.emit(listOf()) }
   }
+
+  private fun copyDatabase(context: Context, onCopyCompleteListener: () -> Unit) {
+    viewModelScope.launch(dispatcherProvider.io()) {
+      try {
+        val passphrase = DBUtils.getEncryptionPassphrase("fhirEngineDbPassphrase")
```

Comment:
Is this the password that encrypts the extracted DB? 

Perhaps we should prompt the user to enter a specific password that can be shared with the troubleshooter for more security. 

If its a heavy lift I'm ok with this being in _v2_ or the version that goes to _main_

Reply:
This passphrase is what FHIR engine uses to encrypt the app's Sqlite DB. The class & methods that handle this in the SDK are private and we had to do a workaround to be able to decrypt the app DB from FHIRCore.
There's a conversation around having the DB decrypted in the SDK as documented [in SDK issue #2360](https://github.com/google/android-fhir/issues/2360). That implementation is what is meant to be used in main.

The password we are using to encrypt zipped DB file after download is `{username}_{first 8 characters of the practitioner_id}`.

For purposes of the release to unblock eCHIS, we had gone with a derivative password to simplify the download and sharing. We can include the prompt for a password in the next iteration using the SDK.